src/common/collisions.js

```diff
@@ -0,0 +1,39 @@
+module.exports = function (rectangle, line) {
+    const minX = rectangle.x - (rectangle.width / 2);
+    const minY = rectangle.y - (rectangle.height / 2);
+    const maxX = rectangle.x + (rectangle.width / 2);
+    const maxY = rectangle.y + (rectangle.height / 2);
+
+    const lineIsLeft = line.originX <= minX && line.endX <= minX;
+    const lineIsBelow = line.originY <= minY && line.endY <= minY;
+    const lineIsRight = line.originX >= maxX && line.endX >= maxX;
+    const lineIsAbove = line.originY >= maxY && line.endY >= maxY;
+
+    if (lineIsLeft || lineIsBelow || lineIsRight || lineIsAbove) {
+        return false;
+    }
+
+    const gradient = (line.endY - line.originY) / (line.endX - line.originX);
+
+    let y = (gradient * (minX - line.originX)) + line.originY;
```

Comment:
These are methods on the linear equation and can use the methods in `public/js/utils.js` (and so that file needs to be re-namespaced)